### PR TITLE
Provide context when serialization fails

### PR DESCRIPTION
**Pull request by dbriones**
Somewhat related to [my previous
PR](https://github.com/aphyr/riemann-clojure-client/pull/16): I noticed that
that I'd occasionally index an event that was impossible to serialize
(that is, with non-String custom attribute values) due to an error in my
Riemann config. The symptoms: inability to remotely query the index, and
a cryptic ClassCastException in the Riemann log.

This made the problem more difficult to diagnose than it ought to have
been. This PR aims to fix that by providing more context when event
serialization fails so that users can more quickly diagnose errors in
their Riemann configurations.

### CODE REVIEW

**Comment by dbriones** [comment]
@aphyr Whoops, I just realized there was a merge conflict on this. All fixed now :)

File: src/riemann/codec.clj [hunk 1]
```diff
@@ -83,7 +83,11 @@
       (when-let [v (get e k)]
         (let [a (Proto$Attribute/newBuilder)]
           (.setKey a (str (when (namespace k) (str (namespace k) \/)) (name k)))
-          (.setValue a v)
+          (try (.setValue a v)
+               (catch java.lang.ClassCastException cce
```
Comment (aphyr) [inline src/riemann/codec.clj]: Out of curiosity, any idea what the performance impact is for doing this catch on every value as opposed to round the entire encode fn?

File: test/riemann/codec_test.clj [hunk 2]
```diff
@@ -102,3 +102,24 @@
         expected (dissoc original :empty-attr)]
     (is (= (msg {:events [expected]})
            (roundtrip {:events [original]})))))
+
+(defn not-serializable-event
```
Comment (aphyr) [inline test/riemann/codec_test.clj]: It looks like this emits both serializable and nonserializable events. Maybe a different function name is in order?

File: test/riemann/codec_test.clj [hunk 3]
```diff
@@ -102,3 +102,24 @@
...
+   :service "someservice"
+   :not-serializable not-serializable-value})
+
+(defn attempt-serialization
+  [value]
+  (try (roundtrip {:events [(not-serializable-event value)]})
+       (catch Throwable t
+         t)))
+
+(deftest attributes-with-non-serializable-values-should-result-in-exception-with-context
```
Comment (aphyr) [inline test/riemann/codec_test.clj]: Maybe use (is (thrown? ex ...)) here?

**Comment by pyr** [comment]
Hi @dbriones, instead of catching i coerced arguments to strings for tags and attribute values

**Comment by dbriones** [comment]
Simpler, and fixes the problem. :+1: Thanks!

**Pull request closed without merging**